### DISCUSSION TITLE
GP sampler hit asset when suggest trial value

### DISCUSSION THREAD

**Discussion by hrntsm** [optuna/optuna#5483] [Q&A]
I was optimizing with GP sampler and it stopped with the following assert.
I couldn't figure out what the problem was beyond the fact that it was an assertion exception, so I commented out this part and ran it, and the optimization continued without any problems.
What is the intent of this assert?

https://github.com/optuna/optuna/blob/efe6f3a0d66a5cfd146a609d8a824d1eb0674ccc/optuna/_gp/optim_mixed.py#L115

This code was called when suggest_int was used to retrieve variables to be used in the trial.

**Comment by nzw0301** [comment]
@contramundum53 could you answer this as the author of corresponding PR?

**Comment by nabenabe0928** [comment]
@hrntsm 
Could you replace it with `np.isclose` and check if it passes?
Otherwise, this is an internal bug.

Basically, in the local search for discrete dimensions, we evaluate acquisition function only on the values in the discrete domains, i.e., `grids`.
The initial point of the discrete line search is merely the given point, i.e., `initial_params[param_idx]`.
Since the start point must also be included in `grids`, this assert sanity-checks whether this holds.
However, as `grids` could be `float` as well, this part should have been `np.isclose` more precisely.

If reproducible, I would like @hrntsm to check once again with `np.isclose` instead of `==`.

**Reply by hrntsm** [reply]
Thanks @nabenabe0928  for your answer.

I rewrite the code as follows and the assertion exception no longer occurs here.

```py
assert np.isclose(initial_params[param_idx],grids[current_choice_i])
```

**Reply by nabenabe0928** [reply]
@hrntsm Thank you for the check!
Then I will publish a PR to fix this issue!

**Comment by nabenabe0928** [comment]
I published the PR, so let me close this discussion!
- https://github.com/optuna/optuna/pull/5484
